openapi3/schema.go

```diff
@@ -1039,6 +1053,37 @@ func (schema *Schema) visitJSONNumber(settings *schemaValidationSettings, value
 		return schema.expectedType(settings, "number, integer")
 	}
 
+	// formats
+	if schemaType == TypeInteger && schema.Format != "" {
+		formatMin := float64(0)
+		formatMax := float64(0)
+		switch schema.Format {
+		case "int32":
+			formatMin = formatMinInt32
+			formatMax = formatMaxInt32
+		case "int64":
+			formatMin = formatMinInt64
+			formatMax = formatMaxInt64
+		default:
+			return ErrSchemaInvalidIntegerFormat
+		}
+		if !(formatMin <= value && value <= formatMax) {
```

Comment:
This is missing a defaults check: `formatMin, formatMax != float64(0), float64(0)`

Reply:
yes, but there is only two values supported and any other value will lead to an error being returned. If you prefer, i may as well not return an error and check the default values instead.

Reply:
Ah sorry I forgot to mention that the OpenAPI specification does not forbid foreign formats so this code cannot just error here. 

Please wrap this error return on the condition of `SchemaFormatValidationDisabled`. See string formats. 

Then this'll get merged!

Reply:
Also please rebase on top of https://github.com/getkin/kin-openapi/pull/584

Reply:
@fenollp : rebased and added defaults check as well as SchemaFormatValidationDisabled